app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,72 @@
+
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: [:index,:show]
+  before_action :find_post, only: [:edit,:update,:show,:destroy]
+  before_action :find_owned_post, only: [:edit,:update,:destroy]
```

Comment:
find_post probably doesn't need to be called for edit, update, destroy, if find_owned_post will overwrite the @post variable.